packages/plugins/validation-cache/src/index.ts

```diff
@@ -50,18 +50,30 @@ export const useValidationCache = (pluginOptions: ValidationCacheOptions = {}):
     onParse({ params, extendContext }) {
       extendContext({ [rawDocumentSymbol]: params.source.toString() });
     },
-    onValidate({ params, context, setResult }) {
-      const key: string = context[rawDocumentSymbol] ?? print(params.documentAST);
-      const cachedResult = resultCache.get(key);
+    onValidate({ params, context, setValidationFn, validateFn }) {
+      // We use setValidateFn over accessing params.rules directly, as other plugins in the chain might add more rules.
+      // This would cause an issue if we are constructing the cache key here already.
+      setValidationFn((...args) => {
+        let ruleKey = '';
+        if (Array.isArray(args[2])) {
```

Comment:
we always have second `arg` here?

Reply:
Well, if we don't have access - we cannot create a cache key. 😅
I assume because of the `useEngine` call, these rules always exist:

https://github.com/n1ru4l/envelop/blob/29326750382ff963619b30ec91241146458a89bd/packages/core/src/plugins/use-engine.ts#L27